scripts/stitching/stitching.py

```diff
@@ -0,0 +1,325 @@
+""" Script for creating a stitched ROI MIBItiff from a folder of FOVs.
+"""
+
+import argparse
+import json
+import os
+import sys
+import time
+
+import numpy as np
+
+from mibidata import mibi_image as mi
+from mibidata import tiff
+from mibitracker.request_helpers import MibiRequests
+
+MAX_TRIES = 5
+
+def combine_entity_by_name(roi_fov_paths, cols, rows, enforce_square, margin):
+    min_col = np.min(cols)
+    min_row = np.min(rows)
+    cols=[v-min_col+1 for v in cols]
+    rows=[v-min_row+1 for v in rows]
+    w = np.max(cols)
+    h = np.max(rows)
+
+    panel = None
+    out_img = None
+    ref_metadata = None
+    for fov_i,roi_fov_path in enumerate(roi_fov_paths):
+        fov_img = tiff.read(roi_fov_path)
+        panel = fov_img.channels
+        fov_img = fov_img.data
+        if ref_metadata is None:
+            ref_metadata = tiff.info(roi_fov_path)
+
+        img_shape = list(np.shape(fov_img))
+        ch_count = np.min(img_shape)
+        shape_2d = img_shape[:-1]
+
+        if out_img is None:
+            out_img = np.zeros(
+                (h*shape_2d[0]+(h-1)*margin["y"], 
+                 w*shape_2d[1]+(w-1)*margin["x"], ch_count),
+                 dtype=fov_img.dtype)
+            out_img_shape = list(np.shape(out_img))
+
+        out_img[((rows[fov_i]-1)*shape_2d[0]+(rows[fov_i]-1)*margin["y"]):
+                (rows[fov_i]*shape_2d[0]+(rows[fov_i]-1)*margin["y"]), 
+                ((cols[fov_i]-1)*shape_2d[1]+(cols[fov_i]-1)*margin["x"]):
+                (cols[fov_i]*shape_2d[1]+(cols[fov_i]-1)*margin["x"]), :] = \
+                    fov_img[:,:,:]
+    
+    if enforce_square:
+        if out_img_shape[0] > out_img_shape[1]:
+            out_img = np.pad(
+                out_img,((0,0),(0,out_img_shape[0]-out_img_shape[1]),(0,0)))
+        elif out_img_shape[0] < out_img_shape[1]:
+            out_img = np.pad(
+                out_img,((0,out_img_shape[1]-out_img_shape[0]),(0,0),(0,0)))
+        
+    return out_img.astype(out_img.dtype, copy=False), panel, \
+        int(max(w*shape_2d[1], h*shape_2d[0])), ref_metadata
+
+
+def stitch_fovs(run_folder, out_folder, session_dict, upload_to_mibitracker,
```

Comment:
to make it clear that `out_folder` can be `None`, can we have it have the default value of `None`?